src/main/java/appeng/util/item/AEItemStack.java

```diff
@@ -106,15 +106,25 @@ public static IAEItemStack fromNBT( final NBTTagCompound i )
 		return item;
 	}
 
-	public static IAEItemStack fromPacket( final ByteBuf data ) throws IOException
+	@Override
+	public void writeToNBT( final NBTTagCompound i )
+	{
+		this.getDefinition().writeToNBT( i );
+		i.setLong( "Cnt", this.getStackSize() );
+		i.setLong( "Req", this.getCountRequestable() );
+		i.setBoolean( "Craft", this.isCraftable() );
+	}
+
+	public static AEItemStack fromPacket( final ByteBuf data ) throws IOException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)